libraries/Microsoft.Bot.Builder.Dialogs/WaterfallDialog.cs

```diff
@@ -69,7 +69,7 @@ public override async Task<DialogTurnResult> DialogResumeAsync(DialogContext dc,
 
             // Increment step index and run step
             var state = dc.ActiveDialog.State;
-            var index = (int)state[StepIndex];
+            var index = Convert.ToInt32(state[StepIndex]);
```

Comment:
Please add a detailed comment here (and/or link to the GitHub Issue) as to why this here. 

Else someone (me!) will change it back to an (int) cast by mistake.

Reply:
Done